### DISCUSSION TITLE
Remix and Sanity v3

### DISCUSSION THREAD

**Discussion by soderlind** [sanity-io/sanity#3642] [Sanity Studio Support] [answered]
Any guides, examples that can help me getting started ?

**Comment by SimeonGriggs** [comment]
Hey @soderlind!

If you're just looking to load Sanity data into a Remix App there's an example in their repository:
* [Remix Sanity Example](https://github.com/remix-run/remix/tree/main/examples/sanity)

Or if you'd like to see a full example with live preview we have a guide on that too:
* [How to build a Remix website with Sanity.io and live preview](https://www.sanity.io/guides/remix-run-live-preview)

But ... if you're after how to _embed_ Sanity Studio v3 _into_ a Remix App, that example is still a work in progress.

**Comment by soderlind** [comment]
@SimeonGriggs I'm after how to embed Sanity Studio v3. I've seen your live preview example, and love to have live preview (with studio v3).

**Comment by soderlind** [comment]
@SimeonGriggs, so something similar to https://github.com/sanity-io/next-sanity#next-sanitystudio-dev-preview, but for remix (i.e. a  github.com/sanity-io/remix-sanity repo ;))

**Comment by SimeonGriggs** [comment, marked as answer]
Hey! I've just created a _super basic_ reproduction of Sanity inside of a Remix app! I intend to keep this up to date as well as v3 is still a moving target :) 

https://github.com/SimeonGriggs/remix-sanity-studio-v3

**Reply by soderlind** [reply]
Thank you, just what I needed :)

**Reply by timbakkum** [reply]
I can't get the styling to work. It seems you have not committed the studio css, which may contain what i'm looking for.

**Reply by timbakkum** [reply]
It seems like SSR support for the styled components need to be added? If I go the /studio using a `Link` i.e. client side navigation, the styles render. Full (re)load doesn't work.

**Reply by SimeonGriggs** [reply]
Got it! I've had this issue before but can't recreate it this time ... have pushed an update to the repo to try and correct it.

**Reply by soderlind** [reply]
Hi @SimeonGriggs, I'm getting 

```
Uncaught TypeError: Cannot read properties of undefined (reading 'projectId')
    at config.ts:19:59
```

Line 19 is: `projectId: typeof document !== "undefined" ? window.ENV.projectId ?? `` : ``,`

**Reply by soderlind** [reply]
@SimeonGriggs a follow up. Following https://bobbyhadz.com/blog/react-cannot-read-property-props-of-undefined I solved the error above doing, in `sanity/config.ts`:

```js
const projectId = () => {
  return typeof window !== "undefined" ? window.ENV.projectId ?? `` : ``;
};
const dataset = () => {
  return typeof window !== "undefined" ? window.ENV.dataset ?? `` : ``;
};
const apiVersion = () => {
  return typeof window !== "undefined" ? window.ENV.apiVersion ?? `` : ``;
};

const projectDetailsBrowser = {
  projectId: projectId(),
  dataset: dataset(),
  apiVersion: apiVersion(),
};
```
